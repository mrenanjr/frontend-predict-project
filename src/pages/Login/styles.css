.container-login {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    height: 100%;
}

.container-login form {
    width: 400px;
    height: 250px;

    background: linear-gradient(180deg, rgba(0, 103, 172, 0) 0%, #0067AC 100%);
    
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    flex-direction: column;

    mix-blend-mode: normal;
    box-shadow: 0px 4px 25px rgba(0, 0, 0, 0.25);
    border-radius: 12px;
}

.container-login form input {
    height: 50px;
    width: 280px;
}

.container-login div p {
    color: #0F084B;
    text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    font-weight: 500;
}

.title {
    font-size: 32px;
    text-align: center;
    margin-bottom: 0px;
}

.title-name {
    font-size: 24px;
    margin-bottom: 25px !important;
}

.container-login form input::placeholder {
    color: white;
    opacity: 0.8;
    font-weight: 200;
  }
  
.container-login form button {
    height: 50px;
    width: 160px;

    color: white;
    font-weight: 400;
    font-size: 18px;

    border: 0.5px outset white;

    cursor: pointer;

    background: linear-gradient(180deg, rgba(53, 68, 79, 0) 40%, #0067AC 100%);;
    border-radius: 8px;

    transition: 0.2s;
}

.container-login form button:hover {
    color: #0067AC;
    background: white;
}

.fadeInDown {
    -webkit-animation-name: fadeInDown;
    animation-name: fadeInDown;
    -webkit-animation-duration: 1s;
    animation-duration: 1s;
    -webkit-animation-fill-mode: both;
    animation-fill-mode: both;
}

@-webkit-keyframes fadeInDown {
    0% {
        opacity: 0;
        -webkit-transform: translate3d(0, -100%, 0);
        transform: translate3d(0, -100%, 0);
    }

    100% {
        opacity: 1;
        -webkit-transform: none;
        transform: none;
    }
}

@keyframes fadeInDown {
    0% {
        opacity: 0;
        -webkit-transform: translate3d(0, -100%, 0);
        transform: translate3d(0, -100%, 0);
    }

    100% {
        opacity: 1;
        -webkit-transform: none;
        transform: none;
    }
}

/* @-webkit-keyframes fadeIn { from { opacity:0; } to { opacity:1; } }
@-moz-keyframes fadeIn { from { opacity:0; } to { opacity:1; } }
@keyframes fadeIn { from { opacity:0; } to { opacity:1; } }

.fadeIn {
  opacity:0;
  -webkit-animation:fadeIn ease-in 1;
  -moz-animation:fadeIn ease-in 1;
  animation:fadeIn ease-in 1;

  -webkit-animation-fill-mode:forwards;
  -moz-animation-fill-mode:forwards;
  animation-fill-mode:forwards;

  -webkit-animation-duration:1s;
  -moz-animation-duration:1s;
  animation-duration:1s;
}

.fadeIn.first {
  -webkit-animation-delay: 0.4s;
  -moz-animation-delay: 0.4s;
  animation-delay: 0.4s;
}

.fadeIn.second {
  -webkit-animation-delay: 0.6s;
  -moz-animation-delay: 0.6s;
  animation-delay: 0.6s;
}

.fadeIn.third {
  -webkit-animation-delay: 0.8s;
  -moz-animation-delay: 0.8s;
  animation-delay: 0.8s;
}

.fadeIn.fourth {
  -webkit-animation-delay: 1s;
  -moz-animation-delay: 1s;
  animation-delay: 1s;
} */