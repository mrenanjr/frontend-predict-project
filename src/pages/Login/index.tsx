import React, { FormEvent } from 'react';
import './styles.css'
import Footer from '../../components/Footer'
import { Input } from 'semantic-ui-react'
import { useHistory } from 'react-router-dom'

const Login = () => {
    const history = useHistory();

    function handleLogin(event: FormEvent) {
        event.preventDefault();

        history.push('/inicio');
    };

    return (
        <>
            <div className="container-login fadeInDown">
                <div >
                    <p className="title">SISAP</p>
                    <p className="title-name">Sistema de Análise Preditiva</p>
                </div>
                <form>
                    <Input className="login-input" icon="user" iconPosition="left" placeholder="Informe seu login único."/>
                    <Input className="password-input" icon="lock" iconPosition="left" placeholder="Informe sua senha."/>
                    <button type="submit" onClick={handleLogin}>Entrar</button>
                </form>
            </div>
            <Footer/>
        </>
    );
}

export default Login;
