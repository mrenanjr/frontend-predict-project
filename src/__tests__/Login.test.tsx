import * as React from 'react';
import { shallow } from 'enzyme';
import Login from '../pages/Login';

test('Login loads corretly', () => {
  const wrapper = shallow(<Login />);

  expect(wrapper).toMatchSnapshot();
});