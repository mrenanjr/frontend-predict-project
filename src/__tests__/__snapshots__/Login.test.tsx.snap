// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login loads corretly 1`] = `
<Fragment>
  <Dimmer
    className=""
  >
    <Loader
      content="Carregando"
      size="large"
    />
  </Dimmer>
  <div
    className="container-login fadeInDown"
  >
    <div>
      <p
        className="title"
      >
        SISAP
      </p>
      <p
        className="title-name"
      >
        Sistema de Análise Preditiva
      </p>
    </div>
    <form>
      <Input
        className="login-input"
        icon="user"
        iconPosition="left"
        onChange={[Function]}
        placeholder="Informe seu login único."
        type="email"
        value=""
      />
      <Input
        className="password-input"
        icon="lock"
        iconPosition="left"
        onChange={[Function]}
        placeholder="Informe sua senha."
        type="password"
        value=""
      />
      <button
        onClick={[Function]}
        type="submit"
      >
        Entrar
      </button>
    </form>
  </div>
  <Footer />
</Fragment>
`;
